### DISCUSSION TITLE
How to deal with Store Lookup measures ?

### DISCUSSION THREAD

**Discussion by EricDallAgnol** [atoti/atoti#592] [Q&A] [answered]
Hi there 👋 ,

I have a question regarding an implementation of a measure I'm working on.

The aim of this measure is to display a field value from a factless table, through a store lookup.
To do so, with the Java API the workflow is basically :
     1.  From the location computed, retrieves members values of several dimensions (let's call them D1 and D2)
     2. Perform a DataStore lookup with as query params the members retrieved above
     
    
For the step 2, I found as a workaround to call the Datastore REST API Endpoint -> [/data/stores/{storeName}?query=](https://artifacts.activeviam.com/documentation/rest/5.11.3/activepivot-datastore.html#data_stores__storeName____query__post)
But I'm not sure if there is a more elegant way to perform it...

Anyway even if I assume step 2 is ok, I need to provide "dynamic" values for the conditions on D1 and D2 in the REST call.
In my case, actually the user will have first to filter on ONE member for D1 and D2, so it would also work if I could get the filter value applied to provide them as query params in my REST call.

Is it possible to have a such logic within Atoti or do I need to build my measure in another way ?

Many thanks 😃

**Comment by mailys-ds** [comment, marked as answer]
Hi 👋,

In atoti, we don't perform datastore lookups as in ActivePivot: we chose to enable them directly with [table joins](https://docs.atoti.io/latest/lib/atoti/atoti.table.html#atoti.Table.join).
So in your case, what I would suggest is to join D1 and D2 in your mapping to your table, like this you will have a reference between your retrieved members and the lookup table. Then you will be able to implement your measure with the columns of your joined tables.

For instance, let's consider a table currency with exchange rates for which you want to do a datastore lookup to convert your measure price in an other currency. In atoti, you will directly retrieve your members D1 and D2 in a table that you will join to your currency table with a mapping `{"currency":"currency"}` to enable the implementation of a calculated measure of the price with your exchange rates.

Let me know if that's clear enough :smile:

**Comment by EricDallAgnol** [comment]
Hi Mailys,

Thanks for your answer! 
Yes that's clear 😄
